backend/app/Http/Controllers/TransactionController.php

```diff
@@ -11,6 +11,38 @@
 
 class TransactionController extends Controller
 {
+    public static function creditTransaction(Account $account, $payload)
+    {
+        $transaction_date = now();
+        $senderData = [
+            'account_id' => $account->id,
+            'user_id' => $account->user_id,
+            'transaction_type' => $payload['transaction_type'],
+            'category' => 'SENDER',
+            'description' => $payload['description'],
+            'starting_balance' => $account->getBalance(),
+            'transaction_amount' => $payload['amount'] * -1,
+            'transaction_date' => $transaction_date,
+        ];
+
+        DB::beginTransaction();
```

Comment:
use DB::transaction(function () {.... https://laravel.com/docs/8.x/database#database-transactions
theres no custom code for the catch code anyways

also 
![image](https://github.com/framgia/sph-flutter/assets/95029803/eeeb2331-cf2b-4c2a-b47a-2a338cf2cf30)